### PR TITLE
[ASCII-2562] Migrate forward eventplatform component to the latest file structure

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Migrate `comp/forwarder/eventplatform` component to the latest file structure

### Motivation

Simplify the component

We replace the need to have dedicated eventplatfom component params, with two separate implementations:
- `comp/forwarder/eventplatform/impl`
- `comp/forwarder/eventplatform/impl-noop`

Previously we needed to provide params to sepcify that we wanted the eventplatform to be disable, now we can use the `noop` implementation

Added benefit for Agents that had a disable event platform component they no longer include the implementation details 

Most of the changes are import updates 😄 

### Describe how to test/QA your changes

Unit tests updated

### CODE REVIEW

**Review by davidor — APPROVED** [review]
Nothing important in container-platform files. Just a couple of updated imports :+1:

**Review by TCheruy — APPROVED** [review]

**Review by vitkyrka — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]
👍🏻 LGTM for files owned by @DataDog/container-integrations.

**Review by remeh — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]

File: pkg/aggregator/demultiplexer_test.go [hunk 1]
```diff
@@ -59,25 +59,23 @@ func TestDemuxForwardersCreated(t *testing.T) {
...
 
-	deps := createDemuxDeps(t, opts, eventplatformimpl.NewDefaultParams())
+	deps := createDemuxDeps(t, opts)
 	demux := deps.Demultiplexer
 
 	require.NotNil(demux)
-	_, found := deps.EventPlatformFwd.Get()
-	require.True(found)
-	_, found = deps.OrchestratorFwd.Get()
+	require.NotNil(deps.SharedForwarder)
```
[the remark this answers is not in the document]
Comment (GustavoCaso) [reply pkg/aggregator/demultiplexer_test.go:66]: Nice catch my friend. Let me fix that

[the remark this answers is not in the document]
Comment (GustavoCaso) [reply pkg/aggregator/demultiplexer_test.go]: Done https://github.com/DataDog/datadog-agent/pull/31471/commits/4225862d1a78fc5934c0c319bae1b6aaf947acc6

**Review by vickenty — APPROVED** [review]

**Review by purple4reina — APPROVED** [review]

**Review by zoedt — APPROVED** [review]

**Review by robertjli — APPROVED** [review]

**Review by akarpz — APPROVED** [review]

File: comp/networkpath/npcollector/npcollectorimpl/npcollectorcomp.go [hunk 2]
```diff
@@ -61,23 +61,17 @@ func newNpCollector(deps dependencies) provides {
 			rdnsQuerier = nooprdnsquerier.NewNone().Comp
 		}
 
-		epForwarder, ok := deps.EpForwarder.Get()
```
[the remark this answers is not in the document]
Comment (GustavoCaso) [reply comp/networkpath/npcollector/npcollectorimpl/npcollectorcomp.go:64]: Correct. No rather than null is will be a noop struct that would do nothing 😄

**Review by misteriaud — APPROVED** [review]

**Pull request merged**